### ISSUE TITLE
Implement official vcpkg port

### ISSUE THREAD

**Issue by gwihlidal** [ispc/ispc#1305]
It would be very nice if an official vcpkg (https://github.com/Microsoft/vcpkg) port could be developed and maintained to simplify integration of ispc into windows cmake build systems.

**Comment by dbabokin** [comment]
I like the idea. Contributions are welcome!

**Comment by cwfitzgerald** [comment]
The biggest issue I had when I was developing my own cmake integration into my build system was that there isn't a easily accessible static url for downloading ispc releases. As of right now the prebuilt releases are on sourceforge which _afaik_ doesn't offer fixed download links. I ended up personally hosting them on my site, but it would be nice if there were machine downloadable links, either through attaching them to the github release, or what have you.

**Comment by dbabokin** [comment]
That's an interesting point. We haven't thought about releases in this prospective, but I totally see your point and agree we need to have download link, which works from scripts/cmake. We'll check what options we have for this.

Are you interested in Windows only (so vcpkg would work for you), or you need all platforms?

**Comment by cwfitzgerald** [comment]
Both me and vcpkg would very much appreciate all three platforms.

**Comment by dbabokin** [comment]
I've tried downloading ispc from script and this works for me:
```
wget https://sourceforge.net/projects/ispcmirror/files/v1.10.0/ispc-v1.10.0-linux.tar.gz/download -O ispc-v1.10.0-linux.tar.gz
```

I.e. the only problem with download is that the wget by default saves to the "download" file, as the link points ends in that name. After renaming (or with -O option) I verified that the file is valid archive.

I'm not sure how it should be handled in CMake, though. @aneshlya may know more on this matter.

**Comment by aneshlya** [comment]
In CMake the easiest way to download file is to use `file(DOWNLOAD <url> <file>)` where `<file>` is a local file name.  However the archive will be downloaded every time when CMake is run.
Probably better way is to use ExternalProject. It extracts archive by default and you can configure when do you want to update already downloaded files.

**Comment by dbabokin** [comment]
We have moved downloads to Github Release, now it should be easier to download in the scripts.

**Comment by aneshlya** [comment]
I see that there were several attempts to add ISPC to vcpkg but they were not merged because of inactivity. It would be a good task for new contributors to give it a try one more time.

**Comment by pbrubaker** [comment]
> I see that there were several attempts to add ISPC to vcpkg but they were not merged because of inactivity. It would be a good task for new contributors to give it a try one more time.

I have been working on creating a couple GitHub templates for ISPC.  I was experimenting with vcpkg tonight and noticed that there was no ispc-compiler package.  It seems there is [precedent for having compiler binaries](https://vcpkg.io/en/package/directx-dxc) in packages so it should be possible.  It would be great to have someone pick this up.

**Comment by pbrubaker** [comment]
> I see that there were several attempts to add ISPC to vcpkg but they were not merged because of inactivity. It would be a good task for new contributors to give it a try one more time.

I agree, it would be great to have someone pick this up.  I have been working on creating a couple GitHub templates for ISPC.  I was experimenting with vcpkg tonight and noticed that there was no ispc-compiler package.  There is [precedent for having compiler binaries](https://vcpkg.io/en/package/directx-dxc) for doing this and it would be awesome to have!
